compiler/rustc_typeck/src/astconv/mod.rs

```diff
@@ -2772,6 +2772,44 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
         Some(r)
     }
 
+    /// make sure that we are in the condition to suggest the blanket implementation, if we are in the
+    /// case of suggest it, use the function `err_reporter` to report the error or suggestion.
+    // FIXME: The copy of TyCtxt it is a bad think in this case?
+    fn maybe_lint_blanket_trait_impl(
+        &self,
+        self_ty: &hir::Ty<'_>,
+        err_reporter: fn(TyCtxt<'tcx>, &hir::Ty<'_>, &str, &str, Vec<(Span, String)>),
+    ) {
+        let tcx = self.tcx();
+        let parent_id = tcx.hir().get_parent_item(self_ty.hir_id);
+        let child_ty = self_ty;
+        if let hir::Node::Item(hir::Item {
+            kind: hir::ItemKind::Impl(hir::Impl { self_ty, of_trait: Some(trait_ref), .. }),
+            ..
+        }) = tcx.hir().get_by_def_id(parent_id)
+        {
+            let hir::TraitRef { path: hir::Path { span, .. }, .. } = trait_ref;
+            let is_local =
+                if let Some(def_id) = trait_ref.trait_def_id() { def_id.is_local() } else { false };
+            if is_local {
+                let blanket_msg = "for local traits use the blanket implementation";
+                let blanket_sugg = vec![
+                    (*span, "<T: ".to_owned()),
+                    (
+                        span.shrink_to_hi(),
+                        format!(
+                            "{}>",
+                            tcx.sess.source_map().span_to_snippet(child_ty.span).unwrap()
+                        ),
+                    ),
+                    (child_ty.span, "T".to_owned()),
+                ];
+                let help_msg = "use blanket implementation";
```

Comment:
I think a help message might be duplicating information, since you already say "use a blanket implementation" once.

Reply:
Mh! and in this case, we should be left the help message empty? I'm not able to see any other help message that avoid the duplicating information

Reply:
In addition, if we go with this https://github.com/rust-lang/rust/pull/97488#discussion_r884165339 I think that the error message should be generalized in the case of the blanket implementation? because in this case there is only one `help_message` for both suggestions, and I don't know in this case how to make a clear distinction.

what do you suggest to do?

Reply:
Sorry, I'm not sure what you're asking here. I think just removing `help_msg` is sufficient.